lib/runner/concurrency/worker-task.js

```diff
@@ -72,6 +73,13 @@ class WorkerTask extends EventEmitter {
     
     const {port1, port2} = new MessageChannel();
     port2.onmessage = ({data: result}) => {
+      if (isString(result)) {
+        try {
+          result = JSON.parse(result);
+          // eslint-disable-next-line no-empty
+        } catch (e) {}
+      }
+
```

Comment:
We don't need this logic here. We can parse everything here:  
https://github.com/nightwatchjs/nightwatch/blob/e7fde68bdf6dd43e9a1c7bcd38263f54dd287bda/lib/reporter/global-reporter.js#L84

Reply:
We can remove the logic to check if it is a child_process.

Reply:
But if we don't parse it here, `result` would be `string` here and hence `result.type` will be undefined below:

https://github.com/nightwatchjs/nightwatch/blob/e7fde68bdf6dd43e9a1c7bcd38263f54dd287bda/lib/runner/concurrency/worker-task.js#L75-L84

That's why I'm parsing it here.